### ISSUE TITLE
\section doesn't support optional argument with apxproof.

### ISSUE THREAD

**Issue by kbauer** [PierreSenellart/apxproof#23]
Normally, sectioning commands allow specifying a shorthand title as optional argument.

With apxproof, this isn't possible for `\section`, while remaining possible for all other sectioning commands.

The failure is silent, as `apxproof`'s version of `\section` will gladly interpret `\section[short]{long}` as the section "[" starting with the text "short]title".

## Example

```
\documentclass{article}
\usepackage{apxproof}

\begin{document}

\section[short title]{title}

\end{document}
```

**Comment by kbauer** [comment]
For reference: The `\section` command is expected to be callable as

    \section{LONG}
    \section[SHORT]{LONG}
    \section*{LONG}

not only for ACM classes, but in pretty much all classes I know of. I'm not sure about `\section*[SHORT]{LONG}` (probably doesn't make sense), but the pull-request would still work fine in that case, since it delegates to `\axp@oldsection`, which would reject `\section*[SHORT]{LONG}` if appropriate.

**Comment by PierreSenellart** [comment]
In the end I see you don't use `xparse`. Thanks for the fix, integrated in 7b86ff7.
